src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,17 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new InvalidDataException("invalid data");
+        } else if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidDataException("problem with login");
+        } else if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidDataException("problem with password");
+        } else if (user.getAge() < 18) {
+            throw new InvalidDataException("problem with age");
+        } else if (storageDao.get(user.getLogin()) == null) {
```

Comment:
The condition `storageDao.get(user.getLogin()) == null` is incorrect for checking if a user already exists. It should be `!= null` to throw an exception if the user already exists in the storage.